platform/api/src/main.rs

```diff
@@ -1,110 +1,221 @@
-                    .sorted_enum_items()
-                    .sorted_fields()
-            )
-        );
-
-        return Ok(());
+impl binary_helper::config::ConfigExtention for ExtConfig {
+    const APP_NAME: &'static str = "scuffle-api";
+
+    fn pre_hook(config: &mut AppConfig) -> anyhow::Result<()> {
+        if config.extra.export_gql {
+            let schema = platform_api::api::v1::gql::schema::<GlobalState>();
+
+            println!(
+                "{}",
+                schema.sdl_with_options(
+                    SDLExportOptions::default()
+                        .federation()
+                        .include_specified_by()
+                        .sorted_arguments()
+                        .sorted_enum_items()
+                        .sorted_fields()
+                )
+            );
+            std::process::exit(0);
+        }
+
+        Ok(())
     }
+}
+
+type AppConfig = binary_helper::config::AppConfig<ExtConfig>;
+
+struct GlobalState {
+    ctx: Context,
+    config: AppConfig,
+    nats: async_nats::Client,
+    jetstream: async_nats::jetstream::Context,
+    db: Arc<sqlx::PgPool>,
+
+    category_by_id_loader: DataLoader<CategoryByIdLoader>,
+    category_search_loader: DataLoader<CategorySearchLoader>,
+    global_state_loader: DataLoader<GlobalStateLoader>,
+    role_by_id_loader: DataLoader<RoleByIdLoader>,
+    session_by_id_loader: DataLoader<SessionByIdLoader>,
+    user_by_id_loader: DataLoader<UserByIdLoader>,
+    user_by_username_loader: DataLoader<UserByUsernameLoader>,
+    user_search_loader: DataLoader<UserSearchLoader>,
+
+    subscription_manager: SubscriptionManager,
+}
 
-    logging::init(&config.logging.level, config.logging.mode)?;
+impl_global_traits!(GlobalState);
 
-    if let Some(file) = &config.config_file {
-        tracing::info!(file = file, "loaded config from file");
+impl common::global::GlobalConfigProvider<ApiConfig> for GlobalState {
+    #[inline(always)]
+    fn provide_config(&self) -> &ApiConfig {
+        &self.config.extra.api
     }
+}
 
-    tracing::debug!("config: {:#?}", config);
+impl common::global::GlobalConfigProvider<TurnstileConfig> for GlobalState {
+    #[inline(always)]
+    fn provide_config(&self) -> &TurnstileConfig {
+        &self.config.extra.turnstile
+    }
+}
 
-    let db = Arc::new(
-        sqlx::PgPool::connect_with(
-            PgConnectOptions::from_str(&config.database.uri)?
-                .disable_statement_logging()
-                .to_owned(),
-        )
-        .await?,
-    );
+impl common::global::GlobalConfigProvider<JwtConfig> for GlobalState {
+    #[inline(always)]
+    fn provide_config(&self) -> &JwtConfig {
+        &self.config.extra.jwt
+    }
+}
 
-    // Sets the similarity threshold for trigram matching to 0.1
-    // Default is 0.3
-    // https://www.cockroachlabs.com/docs/stable/trigram-indexes#comparisons
-    sqlx::query("SET pg_trgm.similarity_threshold = 0.1")
-        .execute(&*db)
-        .await?;
```

Comment:
Where did this go?

Reply:
I was wondering if there is a better way to do this? Perhaps changing the dB configuration so it persists rather than executing it every startup. If the dB restarted the connections would restart and the setting would be lost.

Reply:
Maybe we can include this in a migration?

Reply:
Yeah, if it can be done there or somewhere else.